### DISCUSSION TITLE
Log Device Connect/Disconnect Events and Publish Message Logs to Specific Wildcard Topic

### DISCUSSION THREAD

**Discussion by dineshwar21** [emqx/emqx#14230] [Q&A]
We can use log trace to see log data of publish message in the Specific Wildcard Topic and it as 512MB limit. But can I achieve this any other way. I need both the Connect/Disconnect Events and Publish Message Logs in a same log rotate file. 

It's can be achieve in EMQX itself ? or 

If this functionality isn't available out of the box, what would be the recommended approach to achieve this setup?

**Comment by zmstone** [comment]
I would recommend sending the events to an external services, a few options:

1. Send to Kafka (Enterprise edition)
2. Use 'aggregated' option of the `s3` action to collect Connect/Disconnect and Publish events, and upload to s3. (Enterprise edition)
3. Send to a webhook

If you really want to collect the events in file locally in the broker, you can write a plugin.

**Reply by dineshwar21** [reply]
@zmstone 
Thanks for your time. For this i can use ExHook
